homework/list.js

```diff
@@ -1,79 +1,85 @@
 'use strict';
 
-(function() {
-
-  var _listNoteContent = [];
-  var _wrapper = document.querySelector('#note-list-wrapper');
+(function(exports) {
+    var ListNoteManager = function() {
+        var _listNoteContent = [];
+        var _wrapper = null;
+    };
 
-  function start() {
-    fetchList(function(data) {
-      updateList(data);
-      drawList();
-      preloadFirstNote();
-    });
-    window.addEventListener('click', function(event) {
-      onNoteOpen(event);
-    });
-  }
+    ListNoteManager.prototype = {
+        start() {
+            this._wrapper = document.querySelector('#note-list-wrapper');
+            this.fetchList().then((function(data) {
```

Comment:
You should try to put it in a chain with multiple `then` and `catch`. Please note a `catch` should capture an error and re-throw it after logging it.